src/database/database.py

```diff
@@ -34,32 +73,91 @@ def __init__(self):
         self._db = self._client[cfg['mongo_db']]
 
     def insert(self, col, song_id, doc):
+        """Insert data into the collection
+    
+        Parameters
+        ----------
+        col
+            The collection to be added to
+        song_id
+            id of the song
+        doc
+            Dictionary with data
+            
+        Returns
+        -------
+        int
+            An int of the id
+        """
+
         collection = self._db[col]
         ins = _augment_document(_create_default_document(song_id), doc)
         _id = collection.insert_one(ins).inserted_id
         return _id
 
-    # Gets the newest entry, the other option would be to overwrite it in the insert method
+# Gets the newest entry, the other option would be to overwrite it in the insert method
     def find(self, col, song_id):
+        """Find one instance of the data requested
+    
+        Parameters
+        ----------
+        col
+            The collection to be added to
+        song_id
+            id of the song
+            
+        Returns
+        -------
+        Object
+            Either a None Object or the Object from the database
+        """
         if (self._db[col].count({'song_id': song_id}) > 0):
             res = self._db[col].find({'song_id': song_id}
-                                     ).sort([('last_updated', -1)]
-                                            ).limit(1)
+                                      ).sort([('last_updated', -1)]
+                                             ).limit(1)
             return res[0]
 
         return None
 
     def find_all_by_id(self, name, song_id):
+        """Find all instances of the data requested in the collection by song_id
+    
+        Parameters
+        ----------
+        col
+            The collection to be added to
+        song_id
+            id of the song
+            
+        Returns
+        -------
+        Object list
+            A list of the Objects in the database from a given song_id
+        """
+
         results = []
         for res in self._db[name].find({'song_id': song_id}):
             results.append(res)
         return results
 
     def find_all(self, col):
+        """Find all instances of the data requested in the collection
+    
+        Parameters
+        col
+            The collection to be added to
+            
+        Returns
+        -------
+        Object list
+            A list of the Objects in the database
+        """
+
         results = []
         for r in self._db[col].find():
             results.append(r)
         return results
 
     def close(self):
+
```

Comment:
This line should be removed and you need to add a short description of the method